addons/survey/static/src/js/survey_form.js

```diff
@@ -379,7 +382,30 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 }
             }
 
-            this._prepareSubmitValues(formData, params);
+            const userAnswers = this._prepareSubmitValues(formData, params);
+            var questionIds = $form.find('.js_question-wrapper').map(function() { return this.id; }).get();
+
+            // Show correction before submit if survey scoring type is 'scoring_with_answers_after_page'
+            if (self.scoringAfterPage && params) {
+                if (self.showAnswers || self.showAnswers === undefined) {
+                    const correctAnswers = await self._rpc({
+                        model: 'survey.question',
+                        method: 'get_correct_answers',
```

Comment:
Wondering ... people don't have access to 'survey.question' model, which is logical as otherwise they could just find the right answers using RPC scripts. I had issues trying the incognito mode, my session was always expiring, right at the moment it tries to call 'get_correct_answers'. Aren't we facing a real issue here ? You cannot really do the validation that way, as it means people could fetch the right answers before answering them...

The right move is probably to somehow re-use existing code that does the validation of inputs, and get back wrong / rights answers afterwards, a bit more complicated then I suppose.

Reply:
Hi @tde-banana-odoo :slightly_smiling_face: 
Thanks for the review :smile: 
What do you mean by "people could fetch the right answers before answering them" ? :thinking: 
The answers are only fetched when the user clicks on the 'continue'/'submit' button so after the user answered the page questions and only if the answers are validated by the form validation :thinking: Did I miss something here?

Also thanks for noticing the issue with the access to the 'survey.question' model. :slightly_smiling_face:  I'll look into it! However without rpc calls or having the correct answers from the start of the survey I don't think there is really another way :thinking: 

Thanks for your time!

Reply:
Well what I mean is that you cannot send the "right" answers through any call: otherwise people just have to look at the payload and and get access to the answers. This would break the certification flow as it is not secure anymore. The right way is to send user's answers and says "yarp" or "nay" without sending the value themselves to the frontend.